src/QuantizedMeshIO.cpp

```diff
@@ -108,8 +108,8 @@ constexpr int QUANTIZED_COORDINATE_SIZE = 32767;
 
 static unsigned int scale_coordinate(const double v)
 {
-    const int scaled_v = static_cast<int>(v * QUANTIZED_COORDINATE_SIZE);
     TNTN_ASSERT(v >= 0 && v <= QUANTIZED_COORDINATE_SIZE);
```

Comment:
I can't wrap my head around this patch. scale_coordinate scales a given value in a range [0..1] by multiplying it by a constant. The assertion adds no value where you placed it because value passed supposedly is always  less than the constant. I think it's safe to replace this assertion to be (v >= 0 && v <= 1). This will also allow to remove three assertions from line #382.

Reply:
Makes sense